editoast/src/schema/operational_point.rs

```diff
@@ -22,6 +23,27 @@ pub struct OperationalPoint {
     pub extensions: OperationalPointExtensions,
 }
 
+impl OperationalPoint {
+    
+    pub fn persist(operational_points: &Vec<OperationalPoint>, infrastructure_id: i32, conn: &PgConnection) -> Result<(), Box<dyn ApiError>> {
+
+        use crate::tables::osrd_infra_operationalpointmodel::dsl::*;
+        use crate::diesel::ExpressionMethods;
+        use crate::diesel::RunQueryDsl;
+
+        let mut datas = Vec::new();        
+        operational_points.into_iter().for_each(|operational_point| {
+            datas.push((obj_id.eq(operational_point.get_id().clone()), data.eq(serde_json::to_value(operational_point).unwrap()), infra_id.eq(infrastructure_id)));
+        });
```

Comment:
You could use Iterators and `collect` method to create `datas` . It will allow to remove the mutability of the variable.